src/python_testing/TC_DGSW_2_3.py

```diff
@@ -56,6 +56,18 @@ def is_valid_uint32_value(value):
     def is_valid_str_value(value):
         return isinstance(value, str) and len(value) > 0
 
+    def assert_valid_uint64(self, value, field_name):
+        """Asserts that the value is a valid uint64."""
+        asserts.assert_true(self.is_valid_uint64_value(value), f"{field_name} field should be a uint64 type")
+
+    def assert_valid_uint32(self, value, field_name):
+        """Asserts that the value is a valid uint32."""
+        asserts.assert_true(self.is_valid_uint32_value(value), f"{field_name} field should be a uint32 type")
+
+    def assert_valid_str(self, value, field_name):
```

Comment:
NIT: The `is_valid_str_value` method checks for string being not empty as well, but technically speaking empty string is still string... How about using more verbose name for `is_valid_str_value` and `assert_valid_str` so it will be clear that these methods check for "non-empty-str" without reading doc-string?

Reply:
Acked, since the original test requires check the non-empty string, so I just keep it as it. 

I have a follow-up to move those to the base class, will using more verbose name for is_valid_str_value and assert_valid_str